lib/eventFactory.js

```diff
@@ -73,6 +73,32 @@ function getJobsFromPR(config) {
     return jobsToStart;
 }
 
+/**
+ * Starts the build if the changed file is part of the sourcePaths, or if there is no sourcePaths
+ * @method startBuild
+ * @param  {Object}   config                configuration object
+ * @param  {Object}   config.buildConfig    Build Config to create the build with
+ * @param  {Array}    config.changedFiles   List of files that were changed
+ * @param  {Array}    config.sourcePaths    List of soure paths
+ * @return {Promise}
+ */
+function startBuild(config) {
+    /* eslint-disable global-require */
+    const BuildFactory = require('./buildFactory');
+    const buildFactory = BuildFactory.getInstance();
+    /* eslint-enable global-require */
+    const { buildConfig, changedFiles, sourcePaths } = config;
+    let hasChangeInSourcePaths = true;
+
+    // Only check if sourcePaths exists
+    if (sourcePaths) {
+        hasChangeInSourcePaths = changedFiles.some(file =>
```

Comment:
Should we short circuit if `changedFiles = []`?

Reply:
If it's `[]`, I think it will just be similar to `[].forEach`, which will do nothing. So I don't think we need to short circuit.

Reply:
Okay